examples/complex/dns_spoofing.py

```diff
@@ -28,22 +28,33 @@
 parse_host_header = re.compile(r"^(?P<host>[^:]+|\[.+\])(?::(?P<port>\d+))?$")
 
 
-def request(flow):
-    if flow.client_conn.ssl_established:
-        flow.request.scheme = "https"
-        sni = flow.client_conn.connection.get_servername()
-        port = 443
-    else:
-        flow.request.scheme = "http"
-        sni = None
-        port = 80
-
-    host_header = flow.request.pretty_host
-    m = parse_host_header.match(host_header)
-    if m:
-        host_header = m.group("host").strip("[]")
-        if m.group("port"):
-            port = int(m.group("port"))
-
-    flow.request.host = sni or host_header
-    flow.request.port = port
+class Rerouter:
+    def __init__(self):
+        self.hostHeader = None
+
+    def requestheaders(self, flow):
+        self.hostHeader = flow.request.headers.get('host')
```

Comment:
Can we add a comment here explaining why we do this in the requestheaders call and not down below?